packages/core/src/utils/objectCheck.ts

```diff
@@ -0,0 +1,44 @@
+import type { ProofAttributeInfo } from '../modules/proofs'
+
+export class ObjectCheck {
+  public static objectEquals(x: Map<string, ProofAttributeInfo>, y: Map<string, ProofAttributeInfo>): boolean {
```

Comment:
I think we can make this generic, and not focused on `ProofAttributeInfo`? If it is focused on proofs, it should live in the proofs module, not in the generic utils

Reply:
Made generic by changing from Map<string, ProofAttributeInfo) to Record(string, unknown)
DONE